File: ventes-service/frontend/src/app/models/commande.ts
export interface LigneCommande {
  id?: number;
  designation: string;
  quantite: number;
  prixUnitaire: number;
  total?: number;
}

export interface Commande {
  id?: number;
  numero?: string;
  contactId: number;
  clientNom: string;
  statut: 'BROUILLON' | 'CONFIRME' | 'EN_COURS' | 'LIVRE' | 'ANNULE';
  dateCommande?: string;
  dateLivraison?: string;
  montantTotal?: number;
  notes?: string;
  lignes?: LigneCommande[];
  createdAt?: string;
}

File: ventes-service/backend/src/main/java/com/transfafrica/ventes/model/Commande.java
package com.transfafrica.ventes.model;
import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "commandes")
@Data @NoArgsConstructor @AllArgsConstructor @Builder
public class Commande {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String numero;

    @NotNull
    private Long contactId;

    private String clientNom;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private StatutCommande statut;

    private LocalDate dateCommande;
    private LocalDate dateLivraison;

    @Column(precision = 10)
    private Double montantTotal;

    private String notes;

    @OneToMany(mappedBy = "commande", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<LigneCommande> lignes;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (statut == null) statut = StatutCommande.BROUILLON;
        if (dateCommande == null) dateCommande = LocalDate.now();
    }

    @PreUpdate
    protected void onUpdate() { updatedAt = LocalDateTime.now(); }

    public enum StatutCommande {
        BROUILLON, CONFIRME, EN_COURS, LIVRE, ANNULE
    }
}
